src/types/simulation.d.ts

```diff
@@ -76,3 +77,21 @@ type SimulationDatasetReason = {
   };
   success: boolean;
 };
+
+type SpanAttributes = {
+  arguments: ArgumentsMap;
+  computedAttributes: ArgumentsMap;
+  directiveId?: number;
+};
+
+type SpanId = number;
+
+type Span = {
```

Comment:
What is a span?

Reply:
It's essentially a simulated activity. The database table is called `span` because the simulation engine uses more general terminology (i.e. terminology in the simulation domain as opposed to the activity planning domain). In its simplest form a span is just a simulation result that has a start offset and duration, which we map into an activity for use in the activity planning domain.
You can imagine using the span from the simulation domain outside the planning domain to simulate anything that takes some time. For example imagine simulating a car wash where a span could represent how long it takes to vacuum the car, wash the car, etc.

Reply:
Cool, thanks!